core/src/main/scala/org/apache/spark/ui/jobs/AllJobsPage.scala

```diff
@@ -462,7 +462,10 @@ private[ui] class JobDataSource(
       }
       case "Description" => new Ordering[JobTableRowData] {
         override def compare(x: JobTableRowData, y: JobTableRowData): Int =
-          Ordering.String.compare(x.lastStageDescription, y.lastStageDescription)
+          (x.lastStageDescription, y.lastStageDescription) match {
```

Comment:
You know, this and all the other Orderings here could be simpler. This one can be

`case "Description" => Ordering.by(x => (x.lastStageName, x.lastStageDescription))`

Others like the one above should be `Ordering.by(_.jobData.jobId)`. If that's right I think it may be useful to simplify this pattern everywhere while here.